plumbing/object/file_test.go

```diff
@@ -247,3 +247,29 @@ func (s *FileSuite) TestFileIter(c *C) {
 
 	c.Assert(count, Equals, 1)
 }
+
+func (s *FileSuite) TestFileIterSubmodule(c *C) {
+	st, err := filesystem.NewStorage(fixtures.ByTag("submodule").One().DotGit())
```

Comment:
please, comment what repository are you using for this test, it is very hard to know what should be inside the commit below otherwise.